src/main/java/ml/compute/InputData.java

```diff
@@ -0,0 +1,28 @@
+package ml.compute;
+
+import java.util.Set;
+
+import ml.preprocessor.types.PreprocessedData;
+import ml.preprocessor.types.RawData;
+
+/**
+ * Created by mikeecb on 17/10/2015.
+ */
+public class InputData {
```

Comment:
Not sure why we need a visitor pattern to be honest.

I think the best solution to this is to have something like:

    public interface IInputData<T> {
        public T getData();
    }

And have PreprocessedData and RawData implement this. DAOs should remain the same.